ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +216,36 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)]
+      {:content        (delay (.getInputStream conn))
+       :content-length (delay
+                         (let [len (.getContentLength conn)]
+                           (if (<= 0 len) len)))
+       :directory?     (delay (jar-directory? conn))
+       :last-modified  (delay
+                         (let [last-mod (.getLastModified conn)]
+                           (if-not (zero? last-mod)
+                             (Date. last-mod))))}))
+
+(defmethod resource-info :default
```

Comment:
I wanted to provide more context for the developer here. The default message ("No method in multimethod 'resource-info' for dispatch value: :vfs") doesn't give you the namespace - you have to find `url-response` in the stack and go to the ring source to figure out what you need to do. However, this should be a rare occurrence, so I can remove the default if you prefer.